apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -1,10 +1,192 @@
+      return;
+    }
+
+    // Students get a "come back later" message
+    if (!role_instructor) {
+      res.send(
+        Lti13CourseNavigationNotReady({
+          resLocals: res.locals,
+          courseName,
+        }),
+      );
+      return;
+    }
+
+    // Instructor so lookup their existing information in PL
+    const courses = await selectCoursesWithEditAccess({
+      user_id: res.locals.authn_user.user_id,
+      is_administrator: res.locals.authn_is_administrator,
+    });
+
+    const course_instances: CourseInstanceAuthz[] = [];
+
+    for (const course of courses) {
+      // Only course owners can link
+      if (!course.permissions_course.has_course_permission_own) {
+        continue;
+      }
+
+      const instances = await selectCourseInstancesWithStaffAccess({
+        course_id: course.id,
+        user_id: res.locals.authn_user.user_id,
+        authn_user_id: res.locals.authn_user.user_id,
+        is_administrator: res.locals.authn_is_administrator,
+        authn_is_administrator: res.locals.authn_is_administrator,
+      });
+
+      course_instances.push(...instances);
+    }
+
+    res.send(
+      Lti13CourseNavigationInstructor({
+        resLocals: res.locals,
+        courseName,
+        courses,
+        course_instances,
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const unsafe_course_instance_id = req.body.unsafe_course_instance_id;
+
+    const ltiClaim = new Lti13Claim(req);
+    const authn_lti13_instance_id = req.session.authn_lti13_instance_id;
+
+    // Map passed and auth lti13_instance_id through institution to course instance, or fail
+    await queryOneRowAsync(sql.select_lti13_course_instance_institution, {
+      course_instance_id: unsafe_course_instance_id,
+      lti13_instance_id: req.params.lti13_instance_id,
+      authn_lti13_instance_id,
+    });
+
+    // Check that user is course Owner for this course instance
+    const course_instance = await selectCourseInstanceById(unsafe_course_instance_id);
+    const courses = await selectCoursesWithEditAccess({
+      user_id: res.locals.authn_user.user_id,
+      is_administrator: res.locals.authn_is_administrator,
+    });
+
+    const is_ci_owner = courses.find(
+      (course) =>
+        course.id === course_instance?.course_id &&
+        course.permissions_course.has_course_permission_own,
+    );
+
+    if (ltiClaim.isRoleInstructor() && is_ci_owner) {
+      await queryAsync(sql.insert_lci, {
+        lti13_instance_id: req.params.lti13_instance_id,
+        deployment_id: ltiClaim.deployment_id,
+        context_id: ltiClaim.context?.id,
+        context_label: ltiClaim.context?.label,
+        context_title: ltiClaim.context?.title,
+        course_instance_id: unsafe_course_instance_id,
+      });
+
+      res.redirect(`/pl/lti13_instance/${req.params.lti13_instance_id}/course_navigation?done`);
+    } else {
+      throw new error.HttpStatusError(403, 'Permission denied');
```

Comment:
This is our standard 403 response:
```suggestion
      throw new error.HttpStatusError(403, 'Access denied');
```

Reply:
Addressed in 10bfa667d